Devil fixes

lordpidey opened the pull request:
Numerous small fixes/changes/balances to devils.

Notable changes:
It now requires more souls to increase.  It is now 3,7,12 souls, up from 3,6,9.
Devils now have extra rules, no coercing into signing contracts with violence, and no intentionally physically harming another devil.
Burial garment banishment now works on true devils.  <--- Unreported issue, can I use that fix for the mergetoken?
Renamed all demon files to devil
Updated religious supplies to have burial garments.
Burial garments re-added to box/meta
Increased infernal fireball cooldown
Soulsellers can now use a pitchfork without being harmed.
Added new book crate, which includes a codex gigas to cargo.

lordpidey committed:
Renames demon files to devil.

lordpidey committed:
Adds book crate to cargo, where additional codex gigas (and other random books) can be ordered.

Re-factors bookshelf pre-stocking

lordpidey committed:
You can now banish a true devil that needs burial garments for banishment.  This is done by simply having burial garments in the same tile as it.

lordpidey committed:
Adds burial garments to religious supplies crate.

lordpidey committed:
Increases infernal fireball's cooldown

lordpidey committed:
Soulsellers can now use a pitchfork without being hurt.

lordpidey committed:
Clarifies a banishment ritual.

lordpidey committed:
Re-adds burial garments to box/meta.

KorPhaeron commented on code/game/objects/items/weapons/twohanded.dm:
```diff
@@ -474,7 +474,7 @@
 	if(istype(user, /mob/living))
 		var/mob/living/U = user
-		if(!U.mind.devilinfo)
+		if(U.mind && (!U.mind.devilinfo || (U.mind.soulOwner == U.mind))) //Burn hands unless they are a devil or have sold their soul
```
Did you mean !U.mind.soulowner?

lordpidey replied:
No, by default, you own your own soul.

KorPhaeron commented on code/game/objects/items/weapons/twohanded.dm:
```diff
@@ -484,7 +484,7 @@
 
 /obj/item/weapon/twohanded/pitchfork/demonic/attack(mob/target, mob/living/carbon/human/user)
-	if(!user.mind.devilinfo)
+	if(!user.mind && (!user.mind.devilinfo || (user.mind.soulOwner == user.mind)))
```
!user.mind seems incorrect

lordpidey committed:
Fixes pitchfork hand burning check.

lordpidey committed:
Changes devil upgrade thresholds to 3,7,12 (Up from 3,6,9)
Added new rules for devils to follow (No coercing signing with violence, no intentionally harming other devils directly)

optimumtact commented on code/modules/library/random_books.dm:
```diff
@@ -18,31 +29,36 @@
+		category = null
 	var/c = category? " AND category='[sanitizeSQL(category)]'" :""
-	var/DBQuery/query = dbcon.NewQuery("SELECT * FROM [format_table_name("library")] WHERE isnull(deleted)[c] GROUP BY title ORDER BY rand() LIMIT [book_count];") // isdeleted copyright (c) not me
+	var/DBQuery/query = dbcon.NewQuery("SELECT * FROM [format_table_name("library")] WHERE isnull(deleted)[c] GROUP BY title ORDER BY rand() LIMIT [amount];") // isdeleted copyright (c) not me
```
Why are you touching the library sql?

lordpidey replied:
I renamed a variable.  This code was originally in  /obj/structure/bookcase/random/initialize(), but is now in a completely separate proc.

optimumtact commented on code/modules/library/random_books.dm:
```diff
@@ -18,31 +29,36 @@
+			P.update_icon()
 		return
-
+	if(prob(25))
```
Why?

lordpidey replied:
It was there before.  I assume it's so that occasionally bookshelves aren't stocked with the right books, to further enhance the "Nanotrasen doesn't care" look.

optimumtact replied:
ah I missed the previous in the diff
